src/Quadrature.cpp

```diff
@@ -0,0 +1,1305 @@
+               "which is not divisible by 4", num_azim);
+
+  _num_azim = num_azim;
+}
+
+
+/**
+ * @brief Set the number of polar angles to initialize.
+ * @param num_polar the number of polar angles
+ */
+void Quadrature::setNumPolarAngles(const int num_polar) {
+
+  if (num_polar <= 0)
+    log_printf(ERROR, "Unable to set the number of polar angles to %d "
+               "which is less than or equal to zero", num_polar);
+
+  else if (num_polar % 2 != 0)
+    log_printf(ERROR, "Unable to set the number of polar angles to %d "
+               "which is not divisible by 2", num_polar);
+
+  _num_polar = num_polar;
+}
+
+
+/**
+ * @brief Set the Quadrature's array of sines of each polar angle.
+ * @details This method is a helper function to allow OpenMOC users to assign
+ *          the Quadrature's sin thetas in Python. A user must initialize a
+ *          NumPy array of the correct size (e.g., a float64 array the length
+ *          of the number of polar angles) as input to this function. This
+ *          function then fills the NumPy array with the data values for the
+ *          Quadrature's sin thetas. An example of how this function might be
+ *          called in Python is as follows:
+ *
+ * @code
+ *          sin_thetas= numpy.array([0.05, 0.1, 0.15, ... ])
+ *          polar_quad = openmoc.Quadrature()
+ *          polar_quad.setNumPolarAngles(len(sin_thetas))
+ *          polar_quad.setSinThetas(sin_thetas)
+ * @endcode
+ *
+ * @param sin_thetas the array of sines of each polar angle
+ * @param num_polar the number of polar angles
+ */
+void Quadrature::setThetas(double* thetas, int num_azim_times_polar) {
+
+  if (_num_polar/2 * _num_azim/4 != num_azim_times_polar)
+    log_printf(ERROR, "Unable to set %d thetas for Quadrature "
+               "with %d polar angles and %d azimuthal angles"
+               " in each octant",
+               num_azim_times_polar, _num_polar/2, _num_azim/4);
+
+  /* Deallocate memory if it was allocated previously */
+  if (_thetas != NULL) {
+    for (int i=0; i < _num_azim/2; i++)
+      delete [] _thetas[i];
+    delete [] _thetas;
+  }
+
+  /* Initialize memory for arrays */
+  _thetas = new double*[_num_azim/2];
+  for (int i=0; i < _num_azim/2; i++)
+    _thetas[i] = new double[_num_polar];
+
+  /* Extract sin thetas from user input */
+  int ap=0;
+  for (int a=0; a < _num_azim/4; a++) {
+    for (int p=0; p < _num_polar/2; p++) {
+      if (thetas[ap] < 0. || thetas[ap] > M_PI_2)
+        log_printf(ERROR, "Unable to set theta to %f which is "
+                   "not in the range [0,PI/2]", thetas[ap]);
+
+      _thetas[a][p] = thetas[ap];
+      _thetas[_num_azim/2 - a - 1][p] = thetas[ap];
+      _thetas[a][_num_polar - p - 1] = M_PI - thetas[ap];
+      _thetas[_num_azim/2 - a - 1][_num_polar - p - 1] = M_PI - thetas[ap];
+      ap++;
+    }
+  }
+}
+
+
+/**
+ * @brief Set the Quadrature's array of polar weights
+ * @param weights The polar weights
```

Comment:
Perhaps include a Python code snippet for this method as is done for `Quadrature::setThetas(...)`?